### PR TITLE
add API v2 functionality and info

### PR DESCRIPTION
**Pull request by Bee-lee**
# Description
Adds the possibility to use `/api/v2` prefix along with the usual `/api/v1/`

[README](https://github.com/RedHatInsights/insights-results-smart-proxy/blob/ab18e476a1ce837c1c42e49e013d3fd472bf8805/server/api/README.md) should provide enough information to start

Fixes CCXDEV-5077

## Type of change

Please delete options that are not relevant.

- New feature (non-breaking change which adds functionality)
- This change requires a documentation update
- Documentation update

## Testing steps
`make before_commit` + the usual stuff

## Checklist
* [x] `make before_commit` passes
* [x] updated documentation wherever necessary
* [x] added or modified tests if necessary
* [x] updated schemas and validators in [insights-data-schemas](https://github.com/RedHatInsights/insights-data-schemas) in case of input/output change

### CODE REVIEW

**Review by tisnik — APPROVED** [review]
LGTM and TYVM as well!

**Comment by tisnik** [comment]
@Bee-lee I tried to restart the failing job, but there seems to be glitch in unit test to check OpenAPI response.

**Comment by Bee-lee** [comment]
> @Bee-lee I tried to restart the failing job, but there seems to be glitch in unit test to check OpenAPI response.

yes yes, the PR is not finished yet, I still have some todos to do :) mainly the tests

File: Dockerfile [hunk 1]
```diff
@@ -23,8 +23,10 @@ RUN make build && \
 
 FROM registry.redhat.io/ubi8-minimal
 
-COPY --from=builder /opt/app-root/src/insights-results-smart-proxy .
-COPY --from=builder /opt/app-root/src/openapi.json /openapi/openapi.json
+COPY --from=builder /opt/app-root/src/insights-results-smart-proxy . \
+     --from=builder /opt/app-root/src/server/api/v1/openapi.json /openapi/v1/openapi.json \
+     --from=builder /opt/app-root/src/server/api/v2/openapi.json /openapi/v2/openapi.json \
```
Comment (quarckster) [inline Dockerfile:28]: Extra slash in the end :)

Comment (quarckster) [inline Dockerfile]: I don't think that it will in that way. You should use `COPY` for each new path

Comment (Bee-lee) [reply Dockerfile]: yes, I was unsure of that, TYVM for pointing it out

Comment (Bee-lee) [reply Dockerfile]: fixd, TY!

File: server/endpoints_v2.go [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+	"net/http"
+	"path/filepath"
+
+	httputils "github.com/RedHatInsights/insights-operator-utils/http"
+	"github.com/gorilla/mux"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+const (
+	// ReportEndpointV2 https://issues.redhat.com/browse/CCXDEV-5097
```
Comment (tisnik) [inline server/endpoints_v2.go:27]: it is not needed right now, right?

File: server/endpoints_v2.go [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+	ReportEndpointV2 = "/cluster/{cluster}"
+
+	// ClustersDetail https://issues.redhat.com/browse/CCXDEV-5088
+	ClustersDetail = "/rule​/{rule_id}​/clusters_detail​/"
```
Comment (tisnik) [inline server/endpoints_v2.go:31]: dtto

**Review by tisnik — APPROVED** [review]
LGTM, just 2 quetions...

File: server/endpoints_v2.go (hunk 2 above)
Comment (Bee-lee) [reply server/endpoints_v2.go:27]: yeah, these are the two immediately planned next ones, I just wanted something there to keep the structure

Comment (tisnik) [reply server/endpoints_v2.go:27]: cool TY

**Review by joselsegura — APPROVED** [review]
Only one question: are we going to have a "dev" version of the v2?

**Review by epapbak — APPROVED** [review]
LGTM

**Comment by tisnik** [comment]
> Only one question: are we going to have a "dev" version of the v2?

good question. It depends if it will be needed. For `v2` it would be probably better to add just endpoints that are really used by unified Insights advisor.

**Comment by Bee-lee** [comment]
> Only one question: are we going to have a "dev" version of the v2?
+1 Pavel
I asked briefly on the coffee talk yesterday if we even have debug=true anywhere, because it enables the serving of the dev openapi, so maybe it's just used on CI or somewhere

**Pull request merged**